src/interface.vue

```diff
@@ -28,9 +37,15 @@ export default defineComponent({
 	},
 	emits: ['input'],
 	setup(props, { emit }) {
-		const values = inject<ComputedRef<Record<string, any>>>('values');
-
 		const computedValue = ref('');
+        const relations = useCollectionRelations(props.collection)
+		const values = useDeepValues(
+			inject<ComputedRef<Record<string, any>>>('values')!,
+			relations,
+			props.collection,
+			props.primaryKey,
+			props.template
```

Comment:
Actually there's no needing for it. I tested locally and changes on `primaryKey` are properly handled. I think that's because `primaryKey` and `collection` never changes once the interface is rendered. If we switch from `create` -> `edit` page, the interface is completely re-rendered with new prop values.